repository/src/main/java/org/alfresco/transform/client/registry/CombinedConfig.java

```diff
@@ -267,15 +270,118 @@ public void register(TransformServiceRegistryImpl registry)
+                boolean isPipeline = pipeline != null && !pipeline.isEmpty();
+                boolean isFailover = failover != null && !failover.isEmpty();
+                boolean isOneStepTransform = !isPipeline && !isFailover && !name.equals(LocalPassThroughTransform.NAME);
+
+                if (isPipeline && isFailover)
+                {
+                    throw new IllegalArgumentException("Local transformer " + name +
+                            " cannot have pipeline and failover sections. Read from " + readFrom);
+                }
+
+                // Check to see if the name has been used before.
+                int j = lastIndexOf(name, combinedTransformers, i);
+                String baseUrl = transformAndItsOrigin.baseUrl;
+                if (registry instanceof LocalTransformServiceRegistry)
+                {
+                    baseUrl = ((LocalTransformServiceRegistry)registry).getBaseUrlIfTesting(name, baseUrl);
+                    transformAndItsOrigin = new TransformAndItsOrigin(transformer, baseUrl, readFrom);
+                    combinedTransformers.set(i, transformAndItsOrigin);
+                }
+                if (j >= 0)
+                {
+                    if (baseUrl != null) // If a T-Engine, else it is an override
+                    {
+                        throw new IllegalArgumentException("Local T-Engine transformer names must be unique (" +
+                                name + "). Read from " + readFrom);
+                    }
+
+                    if (isOneStepTransform)
+                    {
+                        // We need to set the baseUrl of the original transform in the one overriding,
+                        // so we can talk to its T-Engine
+                        TransformAndItsOrigin overriddenTransform = combinedTransformers.get(j);
+                        TransformAndItsOrigin overridingTransform = new TransformAndItsOrigin(
+                                transformAndItsOrigin.transformer, overriddenTransform.baseUrl, readFrom);
+                        combinedTransformers.set(i--, overridingTransform);
+                    }
+                    combinedTransformers.remove(j);
+                }
+                else if (isOneStepTransform && baseUrl == null)
+                {
+                    throw new IllegalArgumentException("Local T-Engine transformer " + name +
+                            " must have its baseUrl set in " + LOCAL_TRANSFORMER+name+URL+ " Read from "+ readFrom);
+                }
+            }
+            catch (IllegalArgumentException e)
+            {
+                String msg = e.getMessage();
+                registry.logError(msg);
+                combinedTransformers.remove(i--);
+            }
+        }
+        return combinedTransformers;
+    }
+
+    private static int lastIndexOf(String name, List<TransformAndItsOrigin> combinedTransformers, int toIndex)
+    {
+        for (int j = toIndex-1; j >=0; j--)
```

Comment:
Lists are short (< 100) entries and this is not a frequent or time critical step.